docs: data types (#206)
GreptimeTeam/docs: docs/reference/data-types.md

# Data Types

> GreptimeDB supports following data types:

Reviewer: ```suggestion
GreptimeDB supports following data types:
```

| Type name | Description | Synonyms |
|:-:|:-:| :-:|
|`tinyint`| 8-bit signed small integers between -128~127| |
|`smallint`| 16-bit signed big integers between -32768~32767 | |
|`int`| 32-bit signed integers between -2147483648~2147483647| `integer`|
|`bigint`| 64-bit signed big integers between -9223372036854775808~9223372036854775807| |
|`varchar`|UTF-8 encoded strings|`text`/`string`/`char `|
|`float`|32-bit IEEE754 floating point values ||
|`double`|Double precision IEEE 754 floating point values||
|`boolean`|Boolean values||
|`varbinary`|Variable length binary values||
|`date`|32-bit date values||
|`timestamp`|Timestamp values with millisecond precision||

## Unsigned version of integer types
`int` / `tinyint` / `smallint` / `bigint` also have unsigned version, and there corresponding values ranges are:

- `int unsigned`: 0~4294967295
- `tinyint unsigned`: 0~255
- `smallint unsigned`: 0~65535
- `bigint unsigned`: 0~18446744073709551615

## Variable-sized type limitations

> The max capacities of variable-sized type, like `string` and `varbinary` are determined by their encodings and how storage engine handles them. 

Reviewer 2: ```suggestion
The max capacities of variable-sized type, such as `string` and `varbinary`, are determined by their encodings and how storage engine handles them. 
```

# Data Types

GreptimeDB supports the following data types:

| Type name | Description | Synonyms |
|:-:|:-:| :-:|
|`tinyint`| 8-bit signed small integers between -128~127| |
|`smallint`| 16-bit signed big integers between -32768~32767 | |
|`int`| 32-bit signed integers between -2147483648~2147483647| `integer`|
|`bigint`| 64-bit signed big integers between -9223372036854775808~9223372036854775807| |
|`varchar`|UTF-8 encoded strings|`text`/`string`/`char `|
|`float`|32-bit IEEE754 floating point values ||
|`double`|Double precision IEEE 754 floating point values||
|`boolean`|Boolean values||
|`varbinary`|Variable length binary values||
|`date`|32-bit date values||
|`datetime`|64-bit datetime values||

> |`timestamp`|Timestamp values with millisecond precision||

Reviewer 3: ```suggestion
|`timestamp`| 64-bit timestamp values with millisecond precision||
```
